view/rooms/utils/page.js

```diff
@@ -75,4 +76,73 @@ module.exports = {
       console.log('Cannot inject dynamic contents into iframe.');
     }
   },
+
+  /**
+   * Checks whether a layout object is valid for an array of Bots
+   * @param {Object} layout - Layout object
+   * @param {Array} bots - Bots in the room
+   * @returns {boolean}
+   */
+  botLayoutIsValid(layout, bots) {
+    // Layout does not contain required field
+    if (!layout || !layout.leftColumn || !layout.middleColumn ||
+      !layout.rightColumn) {
+      return false;
+    }
+
+    // Amount of bots in layout does not match amount of bots in room
+    if (layout.leftColumn.length + layout.middleColumn.length +
+      layout.rightColumn.length !== bots.length) {
+      return false;
+    }
+
+    for (const param in layout) {
+      // If a bot in the layout does not actually exist in the room
+      if (layout[param] && layout[param].length
+        && layout[param].every(bot => bots.indexOf(bot) === NEG_ONE)) {
+        return false;
+      }
+    }
+
```

Comment:
I feel that all those validations shouldn't be page.js responsibility, don't know if by design would be OK add into a layout or create something like layout validator. In that way, we would have a more cohesive object, more testable and easy to reuse.

Reply:
@cassiodias That makes sense incase we ever add new functionality to 'layout'. Are you suggesting I create another file in the utils folder called "layout.js" to do all of this or something else?